### PR TITLE
feat: Add getDefaultProcessors to CatalogBuilder

### PR DESCRIPTION
**Pull request by bryce-od**
Signed-off-by: Bryce Larson <bryce.larson@originenergy.com.au>

#### :heavy_check_mark: Checklist

- [ x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend/src/service/NextCatalogBuilder.ts [hunk 1]
```diff
@@ -275,6 +276,30 @@ export class NextCatalogBuilder {
     return this;
   }
 
+  /**
+   * Returns the default list of entity processors. These are responsible for reading,
+   * parsing, and processing entities before they are persisted in the catalog. Changing
+   * the order of processing can give more control to custom processors.
+   *
+   * Consider using with with {@link NextCatalogBuilder#replaceProcessors}
```
Comment (Xantier) [inline plugins/catalog-backend/src/service/NextCatalogBuilder.ts:284]:
```suggestion
   * Consider using with {@link NextCatalogBuilder#replaceProcessors}
```

**Review by freben — APPROVED** [review]
Looks good! Just need to run `yarn build:api-reports` and commit the results.

**Review by Rugvip — APPROVED** [review]
:shipit:

**Comment by suuus** [comment]
Thanks for your contribution @bryce-od 🙌 Please let me know if you’d like some help to contribute further.

**Pull request merged**